### ISSUE TITLE
Decide prioritisation order for methods of passing in translation strings

### ISSUE THREAD

**Issue by vanitabarrett** [alphagov/govuk-frontend#2781]
## What
Decide the priority order if all are provided by the user:
- translation strings via JavaScript
- translation strings via data attributes

## Why
We want to implement two ways of passing translation strings to component JS. We need to know how to handle a situation where a user uses both methods.

## Who needs to work on this
Developers

## Who needs to review this
Developers

## Done when

- [x] We’ve made a decision on the priority order of translation strings
- [x] We’ve written up this decision somewhere
- [x] We should consider documenting this decision within the localisation docs (perhaps the frontend docs). Add a note to https://github.com/alphagov/govuk-frontend-docs/issues/196

**Comment by domoscargin** [comment]
The only wrinkle here for me is that we have two different JS initialisation scenarios:

1. It gets initialised via `initAll`
2. It gets initialised for an individual component

data attributes should definitely take precedence over [1], but [2] is potentially a bit trickier. I guess the deciding factor is that we want to favour data attributes in the Design System.

tldr: data-attributes > JavaScript initialisation

**Comment by querkmachine** [comment]
I don't think we currently have any way of telling whether a JS config option has come from `initAll` or from an individual initialisation; and I'm not sure how simple it would be to give one of those methods priority over the other—as whichever runs first will be the one to configure the component (barring the current issues around multiple initialisations). 

Creating and documenting that complexity is probably not in our interest or useful to users, so I'm in support of data-* attributes always trumping JS configuration.

**Comment by domoscargin** [comment]
OK, it looks like we're all agreed:

**config passed by data attributes always trumps config passed by JavaScript.**
